src/server/storage/storage.js

```diff
@@ -312,6 +313,129 @@ Storage.prototype.makeCommit = function (data, callback) {
     return deferred.promise.nodeify(callback);
 };
 
+Storage.prototype.squashCommits = function (data, callback) {
+    //first we should check if the common ancestor is right
+    var self = this,
+        project,
+        deferred = Q.defer(),
+        fromCommit = data.fromCommit,
+        branchName = undefined,
+        rootHash,
+        toCommit,
+        msg,
+        filterBasedOnReach = function (allCommitItems) {
```

Comment:
right now there are no other use for them and they seems pretty specific, but I am open for discussion